saskatoon/harvest/api.py

```diff
@@ -220,3 +223,176 @@ def list(self, request, *args, **kwargs):
+        total_weight_harvests_per_beneficiary = []
+
+        for beneficiary in beneficiaries:
+            total_weight = self.harvest_yield_queryset.filter(
+                recipient=beneficiary
+            ).aggregate(Sum("total_in_lb"))
+            total_harvests = self.harvest_yield_queryset.filter(
+                recipient=beneficiary
+            ).count()
+
+            if total_weight.get("total_in_lb__sum") is not None:
+                total_weight_harvests_per_beneficiary.append(
+                    (beneficiary, total_harvests, total_weight.get("total_in_lb__sum"))
+                )
+
+        return total_weight_harvests_per_beneficiary
+
+    def get_total_weight_per_picker(self):
+        pickers = Person.objects.all().order_by("first_name")
+        total_weight_harvests_per_picker = []
+
+        for picker in pickers:
+            total_weight = self.harvest_yield_queryset.filter(
+                recipient=picker
+            ).aggregate(Sum("total_in_lb"))
+            total_harvests_leader = self.harvest_queryset.filter(
+                pick_leader__person=picker
+            ).count()
+            total_harvests_rfp = RequestForParticipation.objects.filter(
+                picker=picker
+            ).count()
+            total_harvests_is_accepted = (
+                RequestForParticipation.objects.filter(picker=picker)
+                .filter(is_accepted=True)
+                .count()
+            )
+            total_harvests_recipient = self.harvest_yield_queryset.filter(
+                recipient=picker
+            ).count()
+
+            if total_weight.get("total_in_lb__sum") is not None:
+                total_weight_harvests_per_picker.append(
+                    (
+                        picker,
+                        total_harvests_leader,
+                        total_harvests_rfp,
+                        total_harvests_is_accepted,
+                        total_harvests_recipient,
+                        total_weight.get("total_in_lb__sum"),
+                    )
+                )
+
+        return total_weight_harvests_per_picker
+
+    def get_total_number_beneficiaries(self):
+        beneficiaries = Organization.objects.all()
+        total_number_beneficiaries = []
+
+        for beneficiary in beneficiaries:
+            total = self.harvest_yield_queryset.filter(recipient=beneficiary).aggregate(
+                Sum("total_in_lb")
+            )
+            # ? If beneficiary got some fruit
+            if total.get("total_in_lb__sum") is not None:
+                total_number_beneficiaries.append(beneficiary)
+
+        return len(total_number_beneficiaries)
+
+    @property
+    def seasons(self):
+        """seasons that will be shown in the dropdown list"""
+        seasons = []
+        for harvest in self.queryset.all():
+            if harvest.start_date is not None:
+                seasons.append(harvest.start_date.year)
+        return sorted(list(set(seasons)))
```

Comment:
I think going through all harvests is overkill for what we need, I have already changed the way we get the `season.choices` in `HarvestFilter` to avoid this,

Please try using `HarvestFilter.YEARS` instead:
```
        return Response(
            {
                "season": self.request.query_params.get('season'),
                "seasons": HarvestFilter.YEARS,
                ...
```

While you're at it I would appreciate if you could modify the season filter range in `harvest.filters` so we don't have to hardcode the current year:

```
from datetime import datetime

SEASON_FILTER_RANGE = (2016, datetime.now().year)

```
Thanks!

Reply:
That being said you will now run into issues for years where no there were no harvests:
```
Exception Type: | TypeError
-- | --
int() argument must be a string, a bytes-like object or a number, not 'NoneType'
saskatoon/harvest/api.py, line 256, in get_highlights
```
But I think we should cover the case were `self.get_queryset()` returns an empty queryset (e.g. if we decide to add more filters further down the road it could be possible to get no results)

Reply:
> That being said you will now run into issues for years where no there were no harvests:
> 
> ```
> Exception Type: | TypeError
> -- | --
> int() argument must be a string, a bytes-like object or a number, not 'NoneType'
> saskatoon/harvest/api.py, line 256, in get_highlights
> ```
> 
> But I think we should cover the case were `self.get_queryset()` returns an empty queryset (e.g. if we decide to add more filters further down the road it could be possible to get no results)

@2ynn
I'm thinking of adding a red error message and showing all stats. What do you think?

Reply:
@mina-andajos sure that works, showing no stats at all would also make sense.

BTW I've already pushed the change for YEARS here, should merge soon:
https://github.com/LesFruitsDefendus/saskatoon-ng/pull/286/commits/b937bd1f290f34471c14cb7395e36ef2c6e25e42